xbmc/PlayListPlayer.cpp

```diff
@@ -984,7 +985,17 @@ void PLAYLIST::CPlayListPlayer::OnApplicationMessage(KODI::MESSAGING::ThreadMess
             return;
           }
           if (MUSIC::IsAudio(*item) || IsVideo(*item))
+          {
+            if ((IsVideo(*item) && !g_passwordManager.IsVideoUnlocked()) ||
```

Comment:
On more: Please take a look into implementation of IsVideo() and you will see that it is quite complex, thus expensive, cpu cycle wise. I suggest to calculate it only once and to store the result in a variable.

Reply:
I clearly have learnt something from you - I had just spotted that, thought the same thing and am fixing it now!